### PR TITLE
Avoid using polling for PCAN

### LINKED ISSUE #53: PCAN non-polling recv timeouts
Originally reported by: **Brian Thorne (Bitbucket: [hardbyte](https://bitbucket.org/hardbyte), GitHub: [hardbyte](https://github.com/hardbyte))**

---

Implementing a non-polling `recv` function for the PCAN backend using `PCAN_RECEIVE_EVENT`.

See attached PCAN documentation.

---
- Bitbucket: https://bitbucket.org/hardbyte/python-can/issue/53

### PR DESCRIPTION
**Pull request by christiansandberg**
Utilize events instead of polling when the [pywin32](https://sourceforge.net/projects/pywin32) or [pypiwin32](https://pypi.python.org/pypi/pypiwin32) package is available.

Fixes #53.

### CODE REVIEW

File: can/interfaces/pcan/pcan.py [hunk 1]
```diff
@@ -27,8 +32,8 @@
     timeout_clock = time.clock
 
 # Set up logging
-logging.basicConfig(level=logging.WARNING)
 log = logging.getLogger('can.pcan')
+log.setLevel(logging.WARNING)
```
Comment (hardbyte) [inline can/interfaces/pcan/pcan.py]: IMO we shouldn't be setting the logging level at all outside of our scripts. So :+1: for removing the `basicConfig` but I'd say also remove the `setLevel` call.

File: can/interfaces/pcan/pcan.py [hunk 2]
```diff
@@ -83,6 +88,13 @@ def __init__(self, channel, *args, **kwargs):
         if result != PCAN_ERROR_OK:
             raise PcanError(self._get_formatted_error(result))
 
+        if win32event:
+            self._recv_event = win32event.CreateEvent(None, 0, 0, None)
+            result = self.m_objPCANBasic.SetValue(
```
Comment (hardbyte) [inline can/interfaces/pcan/pcan.py]: Just confirming that the pcan api dictates that this must be a win32event? We can't pass in a more standard Python type such as `threading.Event`?

File: can/interfaces/pcan/pcan.py [hunk 3]
```diff
@@ -83,6 +88,13 @@ def __init__(self, channel, *args, **kwargs):
         if result != PCAN_ERROR_OK:
             raise PcanError(self._get_formatted_error(result))
 
+        if win32event:
```
Comment (hardbyte) [inline can/interfaces/pcan/pcan.py]: nit: `if win32event is not None:`

File: can/interfaces/pcan/pcan.py (hunk 2 above)
Comment (christiansandberg) [inline can/interfaces/pcan/pcan.py]: Unfortunately this is a Windows API thing. Would be nice if Peak could provide a more general way like a blocking function or callbacks like many other interfaces have. pywin32 is quite common on Windows installations but I should definitely add it to the installation instructions.

**Pull request merged**